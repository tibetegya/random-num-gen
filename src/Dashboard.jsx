import React from 'react';
import './assets/styles/dashboard.scss';
import emptyImg from './assets/images/empty-state.svg'
import Navbar from './components/Navbar';
import BoxHead from './components/BoxHead';
import Paginator from './components/Paginator';
import yellowLine from './assets/images/yellow-line.svg';
import greenCircle from './assets/images/green-circle.svg';

class Dashboard extends React.Component {
  constructor(props){
    super(props);
    this.paginateNumbers = this.paginateNumbers.bind(this);
    this.handleGenerate = this.handleGenerate.bind(this);
    this.state = {
      numbers: {},
      showOverview: false
    }
  };

  paginateNumbers(nums) {
    // const { location: { state: { data } } } = this.props;
    const data = {
      max: '0993884838',
      min: '0312332332',
      generated_numbers: [ '0939493123']
    }
    const pages = nums.length / 25;
    let fetchedNumbers = {}
    for(let i = 0; i < pages; i++ ){
      const beginIndex = 25 * i;
      const endIndex = 25 * (i+1);
      fetchedNumbers = {
        ...fetchedNumbers,
        [i]: data.generated_numbers.slice(beginIndex, endIndex)
      }
    };

    console.log('fetchedNumbers', fetchedNumbers);
    this.setState({
      numbers: fetchedNumbers
    });
  }
  handleGenerate() {
    this.setState({ showOverview: true })
  }

  componentDidMount() {
    // const { location: { state: { data } } } = this.props;
    const data = {
      max: '0993884838',
      min: '0312332332',
      generated_numbers: [ '0939493123']
    }
    this.paginateNumbers(data.generated_numbers);
  };

  render (){
    // const { location: { state: { data } } } = this.props;
    const data = {
      max: '0993884838',
      min: '0312332332',
      generated_numbers: [ '0939493123']
    }
    const { numbers, showOverview } = this.state;
    return (
      <React.Fragment>
        <Navbar />
        <div className="wrapper" id="dashboard">
          <img src={yellowLine} alt="line" className="bg" id="yellow-ln"/>
          <img src={greenCircle} alt="line" className="bg" id="green-cir"/>
          <div className="centered">
          {!showOverview ? (
            <div className="empty-state">
              <img src={emptyImg} alt="empty state"/>
              <p>There are no generated Numbers</p>
              <p>press button below to generate</p>
              <button onClick={this.handleGenerate}>Generate new numbers</button>
            </div>
          ): (
            <>
              <div className="overview">
                <BoxHead
                  id="purple-bg"
                  title="Total generated numbers"
                  content={data.generated_numbers.length}
                />
                <BoxHead
                  id="green-bg"
                  title="Max generated number"
                  content={data.max}
                />
                <BoxHead
                  id="orange-bg"
                  title="Min generated number"
                  content={data.min}
                />
              </div>
              <div className="generated">
                <div className="title">Generated numbers</div>
                <div className="tiles">
                  {Object.keys(numbers).length > 0 && numbers['0'].map(number => (<div key={number} className="tile">{number}</div>))}
                </div>
              </div>
              <Paginator
                pageNumber={1}
              />
            </>
          )}
          </div>
        </div>
      </React.Fragment>
    )
  }

}
export default Dashboard;
