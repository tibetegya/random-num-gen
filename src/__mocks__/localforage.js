class LocalForageMock {
	constructor() {
	  this.store = {}
	}
  
	clear() {
	  this.store = {}
    return new Promise((resolve, reject) => {
      resolve(this.store)
    })
	}
  
	getItem(key) {
    return new Promise((resolve, reject) => {
      resolve(this.store[key] || null)
    })
	}
  
	setItem(key, value) {
    this.store[key] = value;
    return new Promise((resolve, reject) => {
      resolve(this.store[key])
    })
	}
  
	removeItem(key) {
	  delete this.store[key]
    return new Promise((resolve, reject) => {
      resolve(true)
    })
	}
}
export default new LocalForageMock();