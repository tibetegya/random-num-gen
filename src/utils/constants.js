 export const CONSTANTS = {
  SIGN_IN: 'SIGN_IN'
};
