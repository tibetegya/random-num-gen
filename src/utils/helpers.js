import localforage from 'localforage';

/**
 * Generates Phone numbers and pasists them to indexDB 
 * @param {*} numberArray // array of existing numbers
 * @param {*} maxNumber // ammount of phone numbers to generate
 */
export const generatedNumbers = (numberArray, maxNumber, data={}) => {
  const owner = localStorage.getItem('companyName');
  while(numberArray.length < maxNumber){
    const random = Math.random();
    const rad = String(random).split('.')[1];
    const gen = '0'+rad.substring(0, 9);
    if (!numberArray.includes(gen)) numberArray.push(gen);
  }
  const sortedArray = randomSort(numberArray)
  return localforage.setItem('data', {
    ...data,
    [owner]: {
      owner,
      numbers: numberArray,
      max: sortedArray[sortedArray.length - 1],
      min: sortedArray[0]
    }
  }).then(addedData => {
    return addedData[owner];
  });
};

/**
 * First checks for existing numbers and then
 * generates and persists phone numbers to indexDB
 * @param {*} maxNumbers 
 */
export function generate (maxNumbers) {
  const owner = localStorage.getItem('companyName');
  return new Promise((resolve, reject) => {
    const storedData = localforage.getItem('data').then(data => {
      let numberArray = [];
      let newMaxNumbers = maxNumbers;

      if(!data) {
        return generatedNumbers(numberArray, newMaxNumbers);
      } else {
        if(data[owner]) {
          if(data[owner].numbers.length > 0) {
            console.log('add existing numbers to new arr');
            numberArray = [ ...data[owner].numbers ]
            newMaxNumbers += numberArray.length
          }
        }
        return generatedNumbers(numberArray, newMaxNumbers, data);
      }
    })
    resolve(storedData)
  })
}

/**
 * Retrives data from indexDB
 */
export const getData = () => {
  const owner = localStorage.getItem('companyName');
  return new Promise((resolve, reject) => {
    const retrievedData = localforage.getItem('data').then(data => {
      if(data) return data[owner];
      return null;
    })
    resolve(retrievedData);
  })
}

/**
 * Sorts an array using Quick sort algorithm
 * @param {*} origArray 
 */
function quickSort(origArray) {
	if (origArray.length <= 1) { 
		return origArray;
	} else {

		var left = [];
		var right = [];
		var newArray = [];
		var pivot = origArray.pop();
		var length = origArray.length;

		for (var i = 0; i < length; i++) {
			if (origArray[i] <= pivot) {
				left.push(origArray[i]);
			} else {
				right.push(origArray[i]);
			}
		}
		return newArray.concat(quickSort(left), pivot, quickSort(right));
	}
};

/**
 * Creates integers from string phone numbers 
 * before sorting and also adds prefixing zeros
 * after sorting.
 * @param {*} stringNumbersArr // input Array with string numbers 
 */
export function randomSort (stringNumbersArr) {
  const intNums = stringNumbersArr.map(num => parseInt(num, 10));
  return quickSort(intNums).map(intNum => {
    let stringNum = String(intNum);
    while(stringNum.length < 10){
      stringNum = '0'+stringNum;
    }
    return stringNum;
  });
}

/**
 * Paginates a given array into pages
 * @param {*} numbers // input numbers array
 */
export const paginateNumbers = (numbers) => {
  const pages = Math.ceil(numbers.length / 25);
  let numberPages = {}
  for(let i = 0; i < pages; i++ ){
    const page = i + 1;
    const beginIndex = 25 * i;
    const endIndex = 25 * (i+1);
    numberPages = {
      ...numberPages,
      [page]: numbers.slice(beginIndex, endIndex)
    }
  };
  return numberPages;
}

/**
 * Creates a CSV file from
 * an input object of phone numbers
 * @param {*} numbersObject 
 */
export const createCsv = (numbersObject) => {
  const rows = Object.values(numbersObject);
  const arrayString = rows.map(e=>e.join(",")).join("\n");
  let contentType = 'data:text/csv;charset=utf-8,'
  return encodeURI(contentType+arrayString);
}

export const getName = () => {
  return localStorage.getItem('companyName') || '';
}

