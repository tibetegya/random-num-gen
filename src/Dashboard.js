import React, { useState, useEffect, useReducer } from 'react';
import './assets/styles/dashboard.scss';
import Navbar from './components/Navbar';
import yellowLine from './assets/images/yellow-line.svg';
import greenCircle from './assets/images/green-circle.svg';
import Loader from './components/Loader';
import EmptyState from './components/EmptyState';
import Overview from './components/Overview';
import Paginator from './components/Paginator';
import GenerateModal from './components/GenerateModal';
import { getData } from './utils/helpers';
import {
  setData,
  handleOptions,
  handleModal,
  handlePaginate,
  handleGenerate
} from './utils/handlers'
class Dashboard extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      loading: false,
      showOverview: false,
      isSignedIn: true,
      optionsOpen:  false,
      modalOpen: false,
      data: {},
      numbers: {},
      csv: '',
      page: '1',
      genAmmount: 10
    }
    this.setData = setData.bind(this)
    this.handlePaginate = handlePaginate.bind(this)
    this.handleGenerate = handleGenerate.bind(this)
    this.handleModal = handleModal.bind(this)
    this.handleOptions = handleOptions.bind(this)
  }
  componentDidMount () {
    if(!localStorage.getItem('companyName')){
      this.setState({ isSignedIn: false })
      this.props.history.push('/');
    } else {
      this.setState({ loading: true })
      getData().then(retrievedData => {
        if(retrievedData){
        this.setData(retrievedData)
        this.setState({ showOverview: true })
        }
        this.setState({ loading: false })
      })
    }
  }
  componentDidUpdate(prevProps, prevState) {
    if (this.state.isSignedIn !== prevState.isSignedIn) {
      this.props.history.push('/');
    }
  }

  render () {
    const { optionsOpen, showOverview, csv, loading, data, numbers, page, modalOpen } = this.state
    return (
      <React.Fragment>
        <Navbar
          isOpen={optionsOpen}
          handleOptions={this.handleOptions}
          handleModal={this.handleModal}
          showOverview={showOverview}
          csvData={csv}
        />
        <div className="wrapper" id="dashboard">
          <img src={yellowLine} alt="line" className="bg" id="yellow-ln"/>
          <img src={greenCircle} alt="line" className="bg" id="green-cir"/>
          <div className="centered">
          {loading ?
            <Loader/> :
              !showOverview ?
                <EmptyState handleModal={this.handleModal}/>
                :
              <React.Fragment>
                <Overview
                  total={data.numbers.length}
                  min={data.max}
                  max={data.min}
                />
                <div className="generated">
                  <div className="title">Generated numbers</div>
                  <div className="tiles">
                    {Object.keys(numbers).length > 0 && numbers[page].map(number => (<div key={number} className="tile">{number}</div>))}
                  </div>
                </div>
                <Paginator
                  pageNumber={page}
                  totalPages={Object.keys(numbers).length}
                  handlePaginate={this.handlePaginate}
                />
              </React.Fragment>
          }
          </div>
        </div>
        <GenerateModal
          handleAction={this.handleGenerate}
          handleModal={this.handleModal}
          isOpen={modalOpen}
        />
    </React.Fragment>
    )
  }
}

export default Dashboard;
