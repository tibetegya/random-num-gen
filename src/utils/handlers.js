import { generate, paginateNumbers, randomSort, createCsv } from './helpers';

export const handleSignIn = function (e) {
  const { target: { name, value } } = e;
  e.preventDefault();
  switch (name) {
    case 'companyName':
    this.setState({ companyName: value })
      break;
    case 'signIn':
      localStorage.setItem('companyName', this.state.companyName)
      this.setState({ isSignedIn: true, modalOpen: false })
      break;
    default:
  };

}

export const handleOptions = function (e) {
  const { target: { name } } = e;
  e.preventDefault();
  switch (name) {
    case 'options':
      this.setState(state => ({
        optionsOpen: !state.optionsOpen
      }))
      break;
    case 'ascending':
      const derivedNumbers = paginateNumbers(randomSort(this.state.data.numbers))
      this.setState({
        numbers: derivedNumbers,
        optionsOpen: false
      })
      break;
    case 'descending':
      const reversedNumbers = paginateNumbers(randomSort(this.state.data.numbers).reverse())
      this.setState({
        numbers: reversedNumbers,
        optionsOpen: false
      })
      break;
    case 'download':
        this.setState({ optionsOpen: false })
      break;
    case 'logout':
      localStorage.removeItem('companyName')
      this.setState({ isSignedIn: false, optionsOpen: false })
      break;
    default:
  };

}


export const handleModal = function (e) {
  const { target: { name } } = e;
  e.preventDefault();
  switch (name) {
    case 'closeModal':
      this.setState({ modalOpen: false, genAmmount: 10 });
      break;
    case 'openModal':
      this.setState({ modalOpen: true, optionsOpen: false });
      break;
    default:
  };
};

export const handlePaginate = function (e) {
  const { numbers, page } = this.state
  e.preventDefault();
  const { target: { name, value } } = e;
  switch(name) {
    case 'paginatorInput':
    if (value !== ''
    && value !== '0'
    && parseInt(value, 10) <= Object.keys(numbers).length) this.setState({ page: value });
      break;
    case 'previous':
      if(page !== '1'){
        const PrevPage = String(parseInt(page, 10) - 1)
        this.setState({ page: PrevPage})
      }
      break;
    case 'next':
    if(page !== String(Object.keys(numbers).length) && Object.keys(numbers).length > 1){
      const nextPage = String(parseInt(page, 10) + 1)
      this.setState({ page: nextPage})
    }
    break;
    default:
  }
}

export const handleGenerate = function (e) {
  let { target: { name, value } } = e;
  const { genAmmount, data } = this.state
  const total = data.numbers ? data.numbers.length : 0
  let ammountValue;

  if(name === 'generateInput' && value !== '') {
    if(parseInt(value.trim(), 10) > (10000 - total)) {
      ammountValue = 0
      e.target.value = '0'
      this.setState({ warning: true });
      setTimeout(() => this.setState({ warning: false }), 4000)
    } else {
      ammountValue = parseInt(value.trim(), 10)
    }
    this.setState({ genAmmount: ammountValue });
  }
  if(name === 'generateBtn') {
    this.setState({
      modalOpen: false,
      showOverview: true,
      loading: true
    })
    generate(genAmmount).then(newData => {
      this.setState({
        modalOpen: false,
        showOverview: true,
        loading: true
      })
      this.setData(newData)
      this.setState({ loading: false })
    })
  }
}

export const setData = function (data) {
  const numberPages = paginateNumbers(data.numbers);
    const csv = createCsv(numberPages)
    this.setState({
      data,
      numbers: numberPages,
      csv
    })
}
