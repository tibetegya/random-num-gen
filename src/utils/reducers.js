import { paginateNumbers, randomSort, createCsv } from './helpers'

function reducer(state, action) {
  switch (action.type) {
    case 'set_data':
    const numberPages = paginateNumbers(action.payload.numbers);
      const csv = createCsv(numberPages)
      return {
        ...state,
        data: action.payload,
        numbers: numberPages,
        csv
      };
    case 'set_numbers':
      return  {
        ...state,
        numbers: action.payload
      };
    case 'ascending_sort':
    const sortedNumbers = randomSort(action.payload)
      return  {
        ...state,
        numbers: paginateNumbers(sortedNumbers)
      };
    case 'descending_sort':
      const reversedNumbers = randomSort(action.payload).reverse()
        return  {
          ...state,
          numbers: paginateNumbers(reversedNumbers)
        };
    default:
  }
}
export default reducer;
