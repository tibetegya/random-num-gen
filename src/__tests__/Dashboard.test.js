import React from 'react';
import { shallow, mount, render } from 'enzyme';
import localforage from 'localforage'
import Dashboard from '../Dashboard';
import { generate } from '../utils/helpers'

describe('Dashboard component test', () => {

  beforeEach(async () => {
    localStorage.setItem('companyName', 'dummy')
  })
  it('should match snapshot', () => {
    const wrapper = mount(<Dashboard
      history={{
        push: jest.fn()
      }}
    />)
    expect(wrapper).toMatchSnapshot();
  })
  it('should redirect to home if there is no company data', () => {
    localStorage.clear()
    localforage.clear()
    const historyMock = jest.fn()
    mount(<Dashboard
      history={{
        push: historyMock
      }}
    />)
    expect(historyMock).toHaveBeenCalled();
  })

  it('should redirect to home if there is no company data', () => {
    const btnMock = jest.fn()
    const wrapper = mount(<Dashboard
      history={{
        push: jest.fn()
      }}
    />)
    wrapper.find('button.options').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'options' }
    });
    expect(btnMock).toHaveBeenCalled();
  })
  it('should return null when data is not present in indexDB', () => {
    localStorage.clear()
    localforage.clear()
    localStorage.setItem('companyName', 'dummy')
    localforage.setItem('invalid_key', {})
    
    const wrapper = mount(<Dashboard
      history={{
        push: jest.fn()
      }}
    />)
    expect(wrapper.state().showOverview).toBeFalsy();
  })
  it('should logout', () => {
    const btnMock = jest.fn()
    const redirectMock = jest.fn()
    const wrapper = mount(<Dashboard
      history={{
        push: redirectMock
      }}
    />)
    wrapper.find('button.logout-btn').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'logout' }
    });
    expect(wrapper.state().isSignedIn).toBeFalsy();
  })
  it('should generate numbers', () => {
    const btnMock = jest.fn()
    const redirectMock = jest.fn()
    const wrapper = mount(<Dashboard
      history={{
        push: redirectMock
      }}
    />)
    wrapper.find('input.generate-input').simulate('change', {
      preventDefault: btnMock,
      target: { name: 'generateInput', value: '100' }
    });
    wrapper.find('button.generate').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'generateBtn' }
    });
    expect(wrapper.state().genAmmount).toEqual(100);
  })
  it('page should remain 1', () => {
    const btnMock = jest.fn()
    const redirectMock = jest.fn()
    const wrapper = shallow(<Dashboard
      history={{
        push: redirectMock
      }}
    />)
    wrapper.instance().handlePaginate({
      preventDefault: btnMock,
      target: { name: 'paginatorInput', value: '7' }
    });
    expect(wrapper.state().page).toEqual('1')
    wrapper.instance().handlePaginate({
      preventDefault: btnMock,
      target: { name: 'previous' }
    });
    expect(wrapper.state().page).toEqual('1')
    wrapper.instance().handlePaginate({
      preventDefault: btnMock,
      target: { name: 'next' }
    });
    expect(wrapper.state().page).toEqual('1')
    wrapper.instance().handlePaginate({
      preventDefault: btnMock,
      target: { name: 'other' }
    });
    expect(wrapper.state().page).toEqual('1')
  })

  it('should open and close modal', () => {
    const btnMock = jest.fn()
    const redirectMock = jest.fn()
    const wrapper = shallow(<Dashboard
      history={{
        push: redirectMock
      }}
    />)
    wrapper.instance().handleModal({
      preventDefault: btnMock,
      target: { name: 'openModal' }
    });
    expect(wrapper.state().modalOpen).toEqual(true)
    wrapper.instance().handleModal({
      preventDefault: btnMock,
      target: { name: 'closeModal' }
    });
    expect(wrapper.state().modalOpen).toEqual(false)
    wrapper.instance().handleModal({
      preventDefault: btnMock,
      target: { name: 'other' }
    });
    expect(wrapper.state().modalOpen).toEqual(false)
  })
  it('should generate numbers', async () => {
    let wrapper;
    const btnMock = jest.fn()
    const inputMock = jest.fn()
    const redirectMock = jest.fn()
    await localforage.setItem('data', {
      dummy: {
        numbers: [ '0000000002', '0000000003', '0000000001' ]
      } 
    }).then( async data => {
      wrapper = await mount(<Dashboard
        history={{
          push: redirectMock
        }}
      />)

    })
    wrapper.find('input.generate-input').simulate('change', {
      preventDefault: inputMock,
      target: { name: 'generateInput', value: '100' }
    });

    await wrapper.find('button.generate').last().simulate('click', {
      preventDefault: btnMock,
      target: { name: 'generateBtn' }
    });
    expect(wrapper.state().genAmmount).toEqual(100)
  })
  it('should warn user upon entering very large number', async () => {
    let wrapper;
    const inputMock = jest.fn()
    const redirectMock = jest.fn()
    await localforage.setItem('data', {
      dummy: {
        numbers: [ '0000000002', '0000000003', '0000000001' ]
      } 
    }).then( async data => {
      wrapper = await mount(<Dashboard
        history={{
          push: redirectMock
        }}
      />)

    })
    wrapper.find('input.generate-input').simulate('change', {
      preventDefault: inputMock,
      target: { name: 'generateInput', value: '100000' }
    });
    expect(wrapper.state().warning).toEqual(true)
  })
  it('should sort ascending numbers', async () => {
    let wrapper;
    const btnMock = jest.fn()
    const redirectMock = jest.fn()
    const ascending = [ '0000000001', '0000000002', '0000000003' ]
    await localforage.setItem('data', {
      dummy: {
        numbers: [ '0000000002', '0000000003', '0000000001' ]
      } 
    }).then( async data => {
      wrapper = await mount(<Dashboard
        history={{
          push: redirectMock
        }}
      />)

    })
    wrapper.find('button.options').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'options' }
    });
    wrapper.find('button.ascending').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'ascending' }
    });
    expect(wrapper.state().numbers['1']).toEqual(ascending)
  })
  it('should sort descending numbers', async () => {
    let wrapper;
    const btnMock = jest.fn()
    const redirectMock = jest.fn()
    const descending = [ '0000000003', '0000000002', '0000000001' ]
    await localforage.setItem('data', {
      dummy: {
        numbers: [ '0000000002', '0000000003', '0000000001' ]
      } 
    }).then( async data => {
      wrapper = await mount(<Dashboard
        history={{
          push: redirectMock
        }}
      />)

    })
    wrapper.find('button.options').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'options' }
    });
    wrapper.find('button.descending').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'descending' }
    });
    expect(wrapper.state().numbers['1']).toEqual(descending)
  })

  it('should click download button', async () => {
    let wrapper;
    const redirectMock = jest.fn()
    const btnMock = jest.fn()
    await localforage.setItem('data', {
      dummy: {
        numbers: [ '0000000002', '0000000003', '0000000001' ]
      } 
    }).then( async data => {
      wrapper = await mount(<Dashboard
        history={{
          push: redirectMock
        }}
      />)

    })

    wrapper.find('button.options').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'options' }
    });
    expect(wrapper.state().optionsOpen).toEqual(true)
    wrapper.find('button.download').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'download' }
    });
    wrapper.find('button.download').simulate('click', {
      preventDefault: btnMock,
      target: { name: 'other' }
    });
    expect(wrapper.state().optionsOpen).toEqual(false)
  })
})
