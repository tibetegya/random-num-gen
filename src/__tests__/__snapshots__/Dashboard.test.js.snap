// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component test should match snapshot 1`] = `ReactWrapper {}`;
